hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -128,7 +132,9 @@ public HoodieMergeHandle(HoodieWriteConfig config, String instantTime, HoodieTab
   public HoodieMergeHandle(HoodieWriteConfig config, String instantTime, HoodieTable<T, I, K, O> hoodieTable,
                            Iterator<HoodieRecord<T>> recordItr, String partitionPath, String fileId,
                            TaskContextSupplier taskContextSupplier, HoodieBaseFile baseFile, Option<BaseKeyGenerator> keyGeneratorOpt) {
-    super(config, instantTime, partitionPath, fileId, hoodieTable, taskContextSupplier);
+    super(config, instantTime, partitionPath, fileId, hoodieTable, taskContextSupplier, false);
+    // The super class constructor is called with preserveMetadata as false
+    this.preserveMetadata = false;
```

Comment:
If parent class already have the flag, let's just use that?

Reply:
parent class doesn't have the flag. HoodieAppendHandle does not use it.